analyses/tcd/notebooks/forecasts/tcd_compare_chirps_arc2.md

```diff
@@ -0,0 +1,260 @@
+        g.fig.subplots_adjust(top=0.9,bottom=0.3,hspace=0.3)
+```
+
+```python
+iso3="tcd"
+```
+
+```python
+chirps_country_processed_dir = Path(data_processed_dir)/iso3/"chirps"
+chirps_country_processed_path = Path(chirps_country_processed_dir)/"monthly"/f"{iso3}_chirps_monthly.nc"
+parameters=config.parameters(iso3)
+adm1_path = Path(os.getenv("AA_DATA_DIR"))/"public"/"raw"/iso3/"cod_ab"/parameters["path_admin1_shp"]
+gdf_adm1=gpd.read_file(adm1_path)
+```
+
+```python
+#months and region of interest for the trigger
+months_sel=[6,7,8,9]
+adm_sel=['Barh-El-Gazel','Batha','Kanem','Lac','Ouaddaï','Sila','Wadi Fira']
+```
+
+```python
+#years that overlap in the two datasets
+#should optimally be set dynamically
+min_year=2000
+max_year=2021
+```
+
+```python
+adm1_col="admin1Name"
+```
+
+```python
+gdf_reg=gdf_adm1[gdf_adm1[adm1_col].isin(adm_sel)]
+```
+
+### Load CHIRPS data
+
+```python
+#load the data
+#when using rasterio, it doesn't read the 'time' coord correctly
+ds_chirps_monthly=xr.load_dataset(chirps_country_processed_path)
+ds_chirps_monthly=ds_chirps_monthly.where(ds_chirps_monthly.time.dt.year.isin(range(min_year,max_year+1)),drop=True)
+da_chirps_monthly=ds_chirps_monthly.precip.rio.set_crs("EPSG:4326",inplace=True)
+```
+
+```python
+#select the region and months of interest
+da_chirps_monthly_reg=da_chirps_monthly.rio.clip(gdf_reg["geometry"])
+da_chirps_monthly_sel=da_chirps_monthly_reg.where(da_chirps_monthly_reg.time.dt.month.isin(months_sel),drop=True)
+```
+
+```python
+#group to yearly data for comparison
+da_chirps_yearly_sel=da_chirps_monthly_sel.groupby(da_chirps_monthly_sel.time.dt.year).sum().rio.clip(gdf_reg["geometry"])
+```
+
+### Load ARC2 data
+
+We first download the ARC2 data. ARC2 is daily data. We also group it to monthly and yearly data to compare with CHIRPS. 
+
+```python
+# #get bounds to define range to download
+# gdf_adm1.total_bounds
+#define class
+arc2 = DrySpells(
+    country_iso3 = iso3,
+    monitoring_start = "1983-01-01",#"2000-01-01",
+    monitoring_end = "2021-11-25",
+    range_x = ("13E", "25E"),
+    range_y = ("7N", "24N")
+)
+
+# #download data, only needed if not downloaded yet
+# arc2.download_data(master=True)
+```
+
+```python
+da_arc = arc2.load_raw_data()
+```
+
+```python
+#rename because .T is taking the transpose so naming it time makes sure there is no confusion
+da_arc=da_arc.rename({"T":"time"})
+#units attrs is very long list of "mm/day" so set to just "mm/day", mainly for plotting
+da_arc.attrs["units"]="mm/day"
+```
+
+```python
+da_arc_country=da_arc.rio.clip(gdf_adm1["geometry"])
+da_arc_country=da_arc_country.where(da_arc_country.time.dt.year.isin(range(min_year,max_year+1)),drop=True)
+```
+
+```python
+#for some reason the resample sets the nan values to zero
+#so clipping again to the country but there should be a better solution for it
+da_arc_monthly=da_arc_country.resample(time='MS',skipna=True).sum().rio.clip(gdf_adm1["geometry"])
```

Comment:
Was double checking but default value of `drop` in `clip` is `True` so that is already being done. The thing is that it will always be a rectangle that is returned (since we work with regular lon/lat grid) and so will keep having some cells on the oustide that are not part of the region and thus should be `NaN`